ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js

```diff
@@ -19,33 +19,41 @@ import { useI18nContext } from '../../../../hooks/useI18nContext';
 const AdvancedGasFeeDefaults = () => {
   const t = useI18nContext();
   const dispatch = useDispatch();
-
   const {
     hasErrors,
     maxBaseFee,
     maxPriorityFeePerGas,
   } = useAdvancedGasFeePopoverContext();
   const advancedGasFeeValues = useSelector(getAdvancedGasFeeValues);
 
-  const updateDefaultSettings = (value) => {
-    if (value) {
+  const [isDefaultSettingsSelected, setDefaultSettingsSelected] = useState(
+    Boolean(advancedGasFeeValues) &&
+      advancedGasFeeValues.maxBaseFee === maxBaseFee &&
+      advancedGasFeeValues.priorityFee === maxPriorityFeePerGas,
+  );
+
+  useEffect(() => {
+    setDefaultSettingsSelected(
+      Boolean(advancedGasFeeValues) &&
+        advancedGasFeeValues.maxBaseFee === maxBaseFee &&
+        advancedGasFeeValues.priorityFee === maxPriorityFeePerGas,
+    );
+  }, [advancedGasFeeValues, maxBaseFee, maxPriorityFeePerGas]);
+
+  const handleUpdateDefaultSettings = () => {
+    if (isDefaultSettingsSelected) {
+      dispatch(setAdvancedGasFee(null));
+      setDefaultSettingsSelected(false);
+    } else {
       dispatch(
         setAdvancedGasFee({
           maxBaseFee,
           priorityFee: maxPriorityFeePerGas,
         }),
       );
-    } else {
-      dispatch(setAdvancedGasFee(null));
+      setDefaultSettingsSelected(true);
     }
   };
```

Comment:
Update will now optimistically set checkbox to true without waiting for state update.